src/org/rascalmpl/library/ParseTree.rsc

```diff
@@ -170,46 +173,51 @@ construct ordered and un-ordered compositions, and associativity groups.
 
 <1> A `prod` is a rule of a grammar, with a defined non-terminal, a list
     of terminal and/or non-terminal symbols and a possibly empty set of attributes.
-  
+
 <2> A `regular` is a regular expression, i.e. a repeated construct.
 
 <3> `priority` means operator precedence, where the order of the list indicates the binding strength of each rule;
 <4> `assoc`  means all alternatives are acceptable, but nested on the declared side;
 <5> `reference` means a reference to another production rule which should be substituted there,
     for extending priority chains and such.
-} 
-data Production 
+<6> `error` means a node produced by error recovery.
+<7> `skipped` means characters skipped during error recovery, always the last child of an `appl` with a `error` production.
+}
+data Production
      = prod(Symbol def, list[Symbol] symbols, set[Attr] attributes) // <1>
      | regular(Symbol def) // <2>
      ;
-     
-data Production 
+
+data Production
      = \priority(Symbol def, list[Production] choices) // <3>
      | \associativity(Symbol def, Associativity \assoc, set[Production] alternatives) // <4>
      | \reference(Symbol def, str cons) // <5>
      ;
 
+data Production
+     = error(Symbol def, Production prod, int dot)
+     | skipped(Symbol symbol);
```

Comment:
I would guess that if a constructor could conceivably be used as a keyword in the future, it is prefixed by a `\` to avoid backwards compatibility issues. Ultimately this is a judgement call. I will prefix both with a `\` just to be on the safe side.